File: lib/app/Models/ProvideOrderDigitalCodesRequest.php
<?php
/**
 * ProvideOrderDigitalCodesRequest
 */
namespace app\Models;

/**
 * ProvideOrderDigitalCodesRequest
 * @description Запрос на передачу ключей цифровых товаров.
 */
class ProvideOrderDigitalCodesRequest {

    /** @var \app\Models\OrderDigitalItemDTO[] $items Список проданных ключей.  Если в заказе есть несколько **одинаковых** товаров (например, несколько ключей к одной и той же подписке), передайте каждый в виде отдельного элемента массива. &#x60;id&#x60; у этих элементов должен быть один и тот же.*/
    public $items = [];

}

File: lib/Models/ProvideOrderDigitalCodesRequest.cpp


#include "ProvideOrderDigitalCodesRequest.h"

using namespace Tiny;

ProvideOrderDigitalCodesRequest::ProvideOrderDigitalCodesRequest()
{
	items = std::list<OrderDigitalItemDTO>();
}

ProvideOrderDigitalCodesRequest::ProvideOrderDigitalCodesRequest(std::string jsonString)
{
	this->fromJson(jsonString);
}

ProvideOrderDigitalCodesRequest::~ProvideOrderDigitalCodesRequest()
{

}

void
ProvideOrderDigitalCodesRequest::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *itemsKey = "items";

    if(object.has_key(itemsKey))
    {
        bourne::json value = object[itemsKey];


        std::list<OrderDigitalItemDTO> items_list;
        OrderDigitalItemDTO element;
        for(auto& var : value.array_range())
        {


            element.fromJson(var.dump());

            items_list.push_back(element);
        }
        items = items_list;


    }


}

bourne::json
ProvideOrderDigitalCodesRequest::toJson()
{
    bourne::json object = bourne::json::object();




    std::list<OrderDigitalItemDTO> items_list = getItems();
    bourne::json items_arr = bourne::json::array();

    for(auto& var : items_list)
    {
        OrderDigitalItemDTO obj = var;
        items_arr.append(obj.toJson());
    }
    object["items"] = items_arr;




    return object;

}

std::list<OrderDigitalItemDTO>
ProvideOrderDigitalCodesRequest::getItems()
{
	return items;
}

void
ProvideOrderDigitalCodesRequest::setItems(std::list <OrderDigitalItemDTO> items)
{
	this->items = items;
}



